### PR TITLE
Fix `runCommand`'s inserts of documents without `_id`s

### LINKED ISSUE #1714: Fix inserts of documents without `_id`s
### What should be done?

When an empty document `bson.D{}` is inserted through `collection.InsertOne()` or `collection.InsertMany()`, the driver itself sets a new unique `_id` for such a document.

However, if such a document is inserted through `db.RunCommand()`, a unique `_id` is not set. 

In this case, FerretDB behaves differently than MongoDB: MongoDB inserts a doc and assigns an `_id`, and FerretDB returns an error.

We need to fix it.

### Where?

See the issue mentioned in the skipped `InsertEmpty` test - https://github.com/FerretDB/FerretDB/pull/1673/files. Remove "skipped" and run the test to see the difference between MongoDB and FerretDB.

### Definition of Done

- compatibility test unskipped.

### PR DESCRIPTION
**Pull request by kropidlowsky**
# Description

Closes #1714 .

- Add `_id`s generation for `runCommand` insert.

**IT DOES NOT WORK FULLY!**
While running `db.runCommand({ insert: "{collection}", documents: [ {v: "abc" } ]})` it fails:
`clientconn/conn.go:164	expected length 95 (marshaled body size) + 16 (fixed marshaled header size) = 111, got 94`.
However, when the FerretDB is run once again the inserted value appears in the collection. 
Thank you in advance for help! 

## Readiness checklist

* [x] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [ ] I marked all done items in this checklist.

### CODE REVIEW

**Comment by w84thesun** [comment]
> `clientconn/conn.go:164	expected length 95 (marshaled body size) + 16 (fixed marshaled header size) = 111, got 94`

That problem usually occurs when we modify the original message and then try to send it to a proxy. As a result of this 
marshal function would fail.

I made the necessary changes to make it work.

File: internal/handlers/pg/msg_insert.go [hunk 1]
```diff
@@ -118,8 +118,17 @@ func insertDocument(ctx context.Context, dbPool *pgdb.Pool, qp *pgdb.QueryParams
 		)
 	}
 
+	toInsert := d
+
+	if !toInsert.Has("_id") {
+		// We have to make a copy to avoid modifying the original document.
```
Comment (rumyantseva) [inline internal/handlers/pg/msg_insert.go:124]: Let's maybe also leave a comment to explain why we can't modify the original document?

File: integration/insert_run_command_compat_test.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ func testInsertRunCommandCompat(t *testing.T, testCases map[string]insertRunComm
 
 					t.Logf("Compat (expected) result: %v", compatRes)
 					t.Logf("Target (actual)   result: %v", targetRes)
-					assert.Equal(t, compatRes, targetRes)
+					assert.Equal(t, compatRes.Map(), targetRes.Map())
```
[the remark this answers is not in the document]
Comment (w84thesun) [reply integration/insert_run_command_compat_test.go]: We returned an incorrect response message before. `Map` was fixing comparison but not the source problem with our code 😅

File: internal/handlers/pg/msg_insert.go (hunk 1 above)
Comment (w84thesun) [reply internal/handlers/pg/msg_insert.go]: Updated comment.

File: internal/handlers/tigris/msg_insert.go [hunk 3]
```diff
@@ -116,6 +116,10 @@ func insertMany(ctx context.Context, dbPool *tigrisdb.TigrisDB, qp *tigrisdb.Que
 
 // insertDocument checks if database and collection exist, create them if needed and attempts to insertDocument the given doc.
 func insertDocument(ctx context.Context, dbPool *tigrisdb.TigrisDB, qp *tigrisdb.QueryParams, doc *types.Document) error {
+	if !doc.Has("_id") {
```
[the remark this answers is not in the document]
Comment (w84thesun) [reply internal/handlers/tigris/msg_insert.go]: Yes, it should. Fixed it.

**Review by w84thesun — APPROVED** [review]
LGTM!

**Pull request merged**